src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/ApplyVQSR.java

```diff
@@ -0,0 +1,628 @@
+ * Each allele will be annotated by its corresponding entry in the AS_FilterStatus INFO field annotation.  Allele-specific VQSLOD and culprit are also carried through from VariantRecalibrator and stored in the AS_VQSLOD and AS_culprit INFO fields, respectively.
+ * The site-level filter is set to the most lenient of any of the allele filters.  That is, if one allele passes, the whole site will be PASS.  If no alleles pass, the site-level filter will be set to the lowest sensitivity tranche among all the alleles.
+ *
+ * Note that the .tranches and .recal files should be derived from an allele-specific run of VariantRecalibrator
+ * Also note that the AS_culprit, AS_FilterStatus, and AS_VQSLOD fields will have placeholder values (NA or NaN) for alleles of a type that have not yet been processed by ApplyRecalibration
+ * The spanning deletion allele (*) will not be recalibrated because it represents missing data. Its VQSLOD will remain NaN and it's culprit and FilterStatus will be NA.
+ *
+ * <h3>Caveats</h3>
+ *
+ * <ul>
+ * <li>The tranche values used in the example above are only meant to be a general example. You should determine the level of sensitivity
+ * that is appropriate for your specific project. Remember that higher sensitivity (more power to detect variants, yay!) comes
+ * at the cost of specificity (more false negatives, boo!). You have to choose at what point you want to set the tradeoff.</li>
+ * <li>In order to create the tranche reporting plots (which are only generated for SNPs, not indels!) Rscript needs to be
+ * in your environment PATH (this is the scripting version of R, not the interactive version).</li>
+ * </ul>
+ */
+@CommandLineProgramProperties(
+        summary = "Apply a score cutoff to filter variants based on a recalibration table",
+        oneLineSummary = " Apply a score cutoff to filter variants based on a recalibration table",
+        programGroup = VariantProgramGroup.class
+)
+public class ApplyVQSR extends VariantWalker {
+
+    protected static final String LOW_VQSLOD_FILTER_NAME = "LOW_VQSLOD";
+    private final double DEFAULT_VQSLOD_CUTOFF = 0.0;
+
+    private boolean foundSNPTranches = false;
+    private boolean foundINDELTranches = false;
+
+    /////////////////////////////
+    // Inputs
+    /////////////////////////////
+    @Argument(fullName="recal_file", shortName="recalFile", doc="The input recal file used by ApplyRecalibration", optional=false)
+    private FeatureInput<VariantContext> recal;
+
+    @Argument(fullName="tranches_file", shortName="tranchesFile", doc="The input tranches file describing where to cut the data", optional=true)
+    private File TRANCHES_FILE;
+
+    /////////////////////////////
+    // Outputs
+    /////////////////////////////
+
+    @Argument(fullName= StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName=StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="The output recal file used by ApplyRecalibration", optional=false)
```

Comment:
Description is wrong -- should be `"The output filtered and recalibrated VCF file in which each variant is annotated with its VQSLOD value"`